### ISSUE TITLE
PlayerNotificationManager pixelated image

### ISSUE THREAD

**Issue by xeinebiu** [androidx/media#994]
### Version

Media3 1.2.1

### More version details

Followed the migration, but I still want to manually control for some use cases the PlayerNotificationManager manually.

Now, when I return a high quality image from **PlayerNotificationManager** 
```kotlin
override fun getCurrentLargeIcon(
      player: Player,
      callback: PlayerNotificationManager.BitmapCallback,
 )
```

The image is pixelated like below.

![image](https://github.com/androidx/media/assets/50703441/2a41acc4-be8c-4fc9-a5e9-9f49332f51b8)

The issue seems to be fixed if I set the artworkUri on the MediaItem, but sadly there is no functionality available where I can set the HTTPS Headers for that Uri therefore that approach doesnt work for me.

### Devices that reproduce the issue

Samsung A52s 5G
Android 14

### Reproducible in the demo app?

Not tested

### Reproduction steps

```kotlin
playerNotificationManager.setMediaSessionToken(mediaSession.sessionCompatToken)
playerNotificationManager.setPlayer(player)
```

return a high quality image from the overriden method

```kotlin
override fun getCurrentLargeIcon(
    player: Player,
    callback: PlayerNotificationManager.BitmapCallback,
): Bitmap? = highQualityImage

```

### Expected result

The largeImage on notification should not be pixelated

### Actual result

The largeImage is pixelated

### Media

![image](https://github.com/androidx/media/assets/50703441/c17dbf7b-7dfa-470d-a2d7-946b94a0e68b)

### Bug Report

- [ ] You will email the zip file produced by `adb bugreport` to android-media-github@google.com after filing this issue.

**Comment by marcbaechinger** [comment]
> The issue seems to be fixed if I set the artworkUri on the MediaItem, but sadly there 
> is no functionality available where I can set the HTTPS Headers for that Uri therefore that approach doesnt work for me.

I think you can implement your own [`BitmapLoader`](https://github.com/androidx/media/blob/release/libraries/common/src/main/java/androidx/media3/common/util/BitmapLoader.java) and [set it when building the session](https://github.com/androidx/media/blob/release/libraries/session/src/main/java/androidx/media3/session/MediaSession.java#L351-L370).

**Comment by xeinebiu** [comment]
> > The issue seems to be fixed if I set the artworkUri on the MediaItem, but sadly there
> > is no functionality available where I can set the HTTPS Headers for that Uri therefore that approach doesnt work for me.
> 
> I think you can implement your own [`BitmapLoader`](https://github.com/androidx/media/blob/release/libraries/common/src/main/java/androidx/media3/common/util/BitmapLoader.java) and [set it when building the session](https://github.com/androidx/media/blob/release/libraries/session/src/main/java/androidx/media3/session/MediaSession.java#L351-L370).

@marcbaechinger thanks for your reply. This looks promising.

I will try to use this loader for now. I think it would be more flexible if we get the MediaItem itself on the callback instead of just the `Uri`. That way we dont have to do a lookup to find the `headers` for the received `Uri`

**Comment by graj-naxtre** [comment]
my issue was resolved after updating the version to 1.3.1 and using exoplayer.mediaMetaData.artworkUri to load image using glide.
`
override fun getCurrentLargeIcon(
                player: Player,
                callback: PlayerNotificationManager.BitmapCallback,
            ): Bitmap? {
                Glide
                    .with(context)
                    .asBitmap()
                    .load(exoPlayer.mediaMetadata.artworkUri)
                    .format(DecodeFormat.PREFER_ARGB_8888)
                    .diskCacheStrategy(DiskCacheStrategy.ALL).override(
                        Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL
                    )
                    .into(
                        object : CustomTarget<Bitmap>() {
                            override fun onResourceReady(
                                resource: Bitmap,
                                transition: Transition<in Bitmap>?,
                            ) {
                                val compressedBitmap = compressBitmap(
                                    resource, 100
                                )
                                callback.onBitmap(compressedBitmap)
                            }

                            override fun onLoadCleared(placeholder: Drawable?) = Unit
                        }
                    )
                return null
            }
            
        private fun compressBitmap(bitmap: Bitmap, quality: Int): Bitmap {
        val stream = ByteArrayOutputStream()
        bitmap.compress(Bitmap.CompressFormat.JPEG, quality, stream)
        return BitmapFactory.decodeStream(ByteArrayInputStream(stream.toByteArray()))
    }
`
